lib/OrdinaryDiffEqRosenbrock/src/rosenbrock_caches.jl

```diff
@@ -1043,10 +946,9 @@ function alg_cache(alg::Rodas5, u, rate_prototype, ::Type{uEltypeNoUnits},
     linsolve = init(linprob, alg.linsolve, alias_A = true, alias_b = true,
         Pl = Pl, Pr = Pr,
         assumptions = LinearSolve.OperatorAssumptions(true))
-    grad_config = build_grad_config(alg, f, tf, du1, t)
-    jac_config = build_jac_config(alg, f, uf, du1, uprev, u, tmp, du2)
-    Rosenbrock5Cache(u, uprev, dense1, dense2, dense3, du, du1, du2, k1, k2, k3, k4,
-        k5, k6, k7, k8,
+    grad_config = build_grad_config(alg, f, tf, dus[2], t)
+    jac_config = build_jac_config(alg, f, uf, dus[2], uprev, u, tmp, dus[3])
```

Comment:
why was this changed to 2 and 3?

Reply:
actually dus had three variables du, du1, and du2 so I kept them in single vector as [du, du1, du2] according to Julia's 1-based indexing du1 would be dus[2] so I went that way